hasktorch/test/TensorSpec.hs

```diff
@@ -16,138 +16,159 @@ import Data.Int
+        asValue (asTensor xx) `shouldBe` xx
+        let xxx = replicate 3 xx
+        asValue (asTensor xxx) `shouldBe` xxx
+    it "TensorLike [Int32]" $ property $
+      \(NonEmpty (x :: [Int32])) -> do
+        asValue (asTensor x) `shouldBe` x
+        toDouble (select (asTensor x) 0 0) `shouldBe` fromIntegral (head x)
+        shape (asTensor x) `shouldBe` [length x]
+        let xx = replicate 5 x
+        asValue (asTensor xx) `shouldBe` xx
+        let xxx = replicate 3 xx
+        asValue (asTensor xxx) `shouldBe` xxx
+    it "TensorLike [Int]" $ property $
+      \(NonEmpty (x :: [Int])) -> do
+        asValue (asTensor x) `shouldBe` x
+        toDouble (select (asTensor x) 0 0) `shouldBe` fromIntegral (head x)
+        shape (asTensor x) `shouldBe` [length x]
+        let xx = replicate 5 x
+        asValue (asTensor xx) `shouldBe` xx
+        let xxx = replicate 3 xx
+        asValue (asTensor xxx) `shouldBe` xxx
+    it "TensorLike [Int64]" $ property $
+      \(NonEmpty (x :: [Int64])) -> do
+        asValue (asTensor x) `shouldBe` x
+        toDouble (select (asTensor x) 0 0) `shouldBe` fromIntegral (head x)
+        shape (asTensor x) `shouldBe` [length x]
+        let xx = replicate 5 x
+        asValue (asTensor xx) `shouldBe` xx
+        let xxx = replicate 3 xx
+        asValue (asTensor xxx) `shouldBe` xxx
+    it "TensorLike [Float]" $ property $
+      \(NonEmpty (x :: [Float])) -> do
+        asValue (asTensor x) `shouldBe` x
+        toDouble (select (asTensor x) 0 0) `shouldBe` realToFrac (head x)
+        shape (asTensor x) `shouldBe` [length x]
+        let xx = replicate 5 x
+        asValue (asTensor xx) `shouldBe` xx
+        let xxx = replicate 3 xx
+        asValue (asTensor xxx) `shouldBe` xxx
+    it "TensorLike [Double]" $ property $
+      \(NonEmpty (x :: [Double])) -> do
+        asValue (asTensor x) `shouldBe` x
+        toDouble (select (asTensor x) 0 0) `shouldBe` realToFrac (head x)
+        shape (asTensor x) `shouldBe` [length x]
+        let xx = replicate 5 x
+        asValue (asTensor xx) `shouldBe` xx
+        let xxx = replicate 3 xx
+        asValue (asTensor xxx) `shouldBe` xxx
+  
+    it "invalid cast of TensorLike a" $ do
+      let x = asTensor (10 :: Int)
+      (dtype x) `shouldBe` Int64
+      (print (asValue x :: Double)) `shouldThrow` anyException
+    it "invalid cast of TensorLike [a]" $ do
+      let x = asTensor ([0..10] :: [Int])
+      (print (asValue x :: [Double])) `shouldThrow` anyException
+  
+    it "lists having different length" $ do
+      (print (asTensor ([[1],[1,2]] :: [[Double]]))) `shouldThrow` anyException
+    it "cast of Tensor" $ do
+      let x = asTensor ([0..10] :: [Int])
+      (dtype (toType Float x)) `shouldBe` Float
 
+  describe "indexing" $ do
+    it "pick up a value" $ do
+      let x = asTensor ([[[0,1,2],[3,4,5]],[[6,7,8],[9,10,11]]] :: [[[Int]]])
+      show (x @@ (1::Int,0::Int,2::Int)) `shouldBe` "Tensor Int64 []  8"
```

Comment:
I think it would be better to compare tensor values and properties directly rather than relying on the Show instance here

Reply:
Yeah, I've replaced it.